sdks/go/pkg/beam/options/jobopts/options_test.go

```diff
@@ -0,0 +1,122 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package jobopts
+
+import (
+	"context"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/core"
+)
+
+func TestGetEndpoint(t *testing.T) {
+	address := "localhost:8099"
+	Endpoint = &address
+	v, err := GetEndpoint()
+	if err != nil {
+		t.Fatalf("unexpected error GetEndpoint(): %v", err)
+	}
+	if v != address {
+		t.Errorf("GetEndpoint() = %v, want %v", v, address)
+	}
+}
+
+func TestGetEndpoint_Bad(t *testing.T) {
+	address := ""
+	Endpoint = &address
+	expectedErr := "no job service endpoint specified. Use --endpoint=<endpoint>"
+	if _, err := GetEndpoint(); err == nil {
+		t.Errorf("GetEndpoint() executed incorrectly, expected error: %v", expectedErr)
+	}
+}
+
+func TestGetJobName(t *testing.T) {
+	tests := []struct {
+		jobname  string
+		wantName string
+	}{
+		{
+			"",
+			"go-job-",
+		},
+		{
+			"go-job-example",
+			"go-job-example",
+		},
+	}
+	for _, test := range tests {
+		JobName = &test.jobname
+		if got := GetJobName(); !strings.Contains(got, test.wantName) {
+			t.Errorf("GetJobName() = %v, want %v", got, test.wantName)
+		}
+	}
+}
+
+func TestGetEnvironamentUrn(t *testing.T) {
+	tests := []struct {
+		env string
+		urn string
+	}{
```

Comment:
Since this is for coverage I think you should test each type of possible env to use as input. Or at the very least, add "DOCKER" there (since I see only "docker" and "loopback" are actually documented in the flag description).